### PR TITLE
Adds profiles component

### PR DESCRIPTION
**Pull request by swiftdiaries**
Link to ksonnet components this PR changes:
[Profiles](https://github.com/kubeflow/kubeflow/tree/master/kubeflow/profiles)

Mapping ksonnet parameters to kustomize is kind of hard. For installing kubeflow to a custom namespace, with ksonnet we set an app level `ks env` with the namespace.

Kustomize does support a global namespace but it's unable to set it inside spec fields like in ambassador's deployment.

Opened an issue in kustomize for this: https://github.com/kubernetes-sigs/kustomize/issues/854
I'm currently investigating how to add multiple parameters like we have in ksonnet to kubeflow manifests.

With respect to a ksonnet package having multiple prototypes, I'm currently treating them as separate manifests. See pytorch and pytorch-job in PR(#5)
This works well for pytorch-job because the only parameter is the image. And that can be set in straight-forward way using `kustomize edit set image <image name>` from inside the manifest folder.

/cc @jlewi @kkasravi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/2)

### CODE REVIEW

**Comment by jlewi** [comment]
Marking this WIP because its blocked on other PRs. When the other PRs are merged and this one is updated remove WIP (this way I'll know its ready for another look).

**Comment by swiftdiaries** [comment]
/assign @jlewi 
Ready for review.

**Comment by jlewi** [comment]
Can you please update the PR description to include the following

* Links to the ksonnet packages these are based on
* How you are mapping whatever customizability exists in the ksonnet package to the Kubeflow package?
   * In particular, does the ksonnet package have multiple parameters? If so how are they mapped to kustomize?
   * If there are multiple ksonnet prototypes how are these mapped to kustomize?

Also going forward it might be good to do a single package per PR.

**Comment by swiftdiaries** [comment]
Ok, that makes sense. Will update the PR with comments and split out future PRs into individual components.

**Comment by jlewi** [comment]
Thanks.

File: README.md [hunk 1]
```diff
@@ -5,6 +5,16 @@ A repository for Kustomize manifests
...
+
+* Ambassador
+
+* Argo
+
+* Jupyter
+
+* Profiles
+
 ## Basic Usage
```
Comment (kkasravi) [inline README.md]: We should probably add a test framework - https://github.com/kubernetes-sigs/kustomize#a-great-report-is-a-bug-reproduction-test

**Comment by swiftdiaries** [comment]
Updated the PR comments. And added parameters from ksonnet to the README to track

**Comment by jlewi** [comment]
Sorry for the slow reply. Since this is one of the first kustomize package conversions I'd like to take a close look. I'm a bit swamped today but I will do my best to make time to look at it.

**Comment by swiftdiaries** [comment]
Ok sounds good! Thanks :)

**Comment by jlewi** [comment]
We are likely moving away from Ambassador so i don't think its super critical to have a convenient way to set the namespace for Ambassador.

**Comment by swiftdiaries** [comment]
Ohh cool. I'll modify the README and add a separate doc for migration. 
For number of replicas, the current usage is that you have an overlay defined for it.
For common edits/variants, I think it'll be better if we provide manifests and overlays OTB. Docs that point to how you use/set those variants. 

Ok noted. Will remove all references to JupyterHub.

**Comment by jlewi** [comment]
@swiftdiaries Thanks is this ready for another look or are you still working on it?

**Comment by swiftdiaries** [comment]
I’m still working on this. Ill get some commits in the evening.

**Comment by swiftdiaries** [comment]
@jlewi This is ready for another look.
I'm currently referencing the profiles controller kustomization from within my fork. Once https://github.com/kubeflow/kubeflow/pull/2692 merges, I can replace it in the next commit.

**Comment by jlewi** [comment]
/lgtm
/approve
Please file an issue to fix the reference to the kustomize package.

**Pull request merged**